python/ray/tune/tune_config.py

```diff
@@ -0,0 +1,43 @@
+from dataclasses import dataclass
+from typing import Optional
+
+from ray.tune.schedulers import TrialScheduler
+from ray.tune.suggest import Searcher
+from ray.util import PublicAPI
+
+
+# TODO(xwjiang): Change to alpha
+@dataclass
+@PublicAPI(stability="beta")
+class TuneConfig:
+    """Tune specific config, related to how hyperparameter tuning algorithms
+    are set up.
+
+    Args:
+        metric: Metric to optimize. This metric should be reported
+            with `tune.report()`. If set, will be passed to the search
+            algorithm and scheduler.
+        mode: Must be one of [min, max]. Determines whether objective is
+            minimizing or maximizing the metric attribute. If set, will be
+            passed to the search algorithm and scheduler.
+        search_alg: Search algorithm for optimization. Default to
+            random search.
+        scheduler: Scheduler for executing
+            the experiment. Choose among FIFO (default), MedianStopping,
+            AsyncHyperBand, HyperBand and PopulationBasedTraining. Refer to
+            ray.tune.schedulers for more options.
+        num_samples: Number of times to sample from the
+            hyperparameter space. Defaults to 1. If `grid_search` is
+            provided as an argument, the grid will be repeated
+            `num_samples` of times. If this is -1, (virtually) infinite
+            samples are generated until a stopping condition is met.
```

Comment:
How about failure handling behavior? That seems like an important config to support initially.

Reply:
max_failures, fail_fast?
Should them be RunConfig instead?
The way I draw the line is TuneConfig is just tune algorithm specific.
Anything that can be shared with train needs to go to RunConfig. wdyt?

Also in RunConfig, do we prefer to have a FailureConfig nested under RunConfig or we just put the two there flat. Would like to know your preference.

Reply:
Added both to TuneConfig. PTAL.

Reply:
The line makes sense to me. Happy with including it in the tune config.

Reply:
Yeah, we are thinking about how to keep this kind of configuration slim and one proposal is to demote some knobs to env variables.

Reply:
Removing max_failures would make our fault tolerance unconfigurable right?

How about we just tag as alpha, keep max_failures, and change it accordingly for beta.

Reply:
I am actually ok with `max_failures`. `fail_fast` is a bit vague.